include/library/spdm_requester_lib.h

```diff
@@ -197,6 +197,41 @@ libspdm_return_t libspdm_challenge_ex(void *spdm_context, void *reserved,
                                       void *opaque_data,
                                       size_t *opaque_data_size);
 
+/**
+ * This function sends CHALLENGE to authenticate the device based upon the key in one slot.
+ *
+ * This function verifies the signature in the challenge auth.
+ *
+ * If basic mutual authentication is requested from the responder,
+ * this function also performs the basic mutual authentication.
+ *
+ * @param  spdm_context           A pointer to the SPDM context.
+ * @param  reserved               Reserved for session_id and is ignored.
+ * @param  slot_id                The number of slot for the challenge.
+ * @param  measurement_hash_type  The type of the measurement hash.
+ * @param  measurement_hash       A pointer to a destination buffer to store the measurement hash.
+ * @param  slot_mask              A pointer to a destination to store the slot mask.
+ * @param  requester_nonce_in     A buffer to hold the requester nonce (32 bytes) as input, if not NULL.
+ * @param  requester_context      A buffer to hold the requester context (8 bytes) as input, if not NULL.
```

Comment:
If `requester_context` is non-NULL and the negotiated version is <= 1.2 then it should return `LIBSPDM_STATUS_OVERRIDDEN_PARAMETER`. Or maybe make a new warning `LIBSPDM_STATUS_IGNORED_PARAMETER`.

Reply:
I am not sure if that is a good idea, because it adds burden to the integrator to understand the version.

SPDM 1.1/1.2 already added new extension, and we don't return such overridden or ignored parameter. Why we need to add in SPDM 1.3?

Reply:
The header documentation should then state that this parameter is ignored if the negotiated version is <= 1.2.

Reply:
OK I will update the comment.